message/pool/message_test.go

```diff
@@ -0,0 +1,30 @@
+package pool
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func TestMessageSetPath(t *testing.T) {
+	msg := NewMessage()
+	size := 4
+	for i := 0; i < 8; i++ {
+		buffer := make([]byte, size)
+		buffer[0] = '/'
+		_, err := rand.Read(buffer[1:])
+		require.NoError(t, err)
+		inPath := string(buffer)
+		msg.SetPath(inPath)
```

Comment:
SetPath crashes for paths longer than 255 chars,

[https://github.com/plgd-dev/go-coap/blob/c3e8d8de2f5c3a0cd69a2811200308613e6243bc/message/pool/message.go#L94-L99]

Reply:
if `used < 0` then the error `message.ErrTooSmall` is incorrect. We need to introduce error `message.ErrTooLong` for this case.

Reply:
Should be fixed now. 

I've also found an edge case when the path had duplicated '/' characters. I've fixed that too and added tests.